lib/peopleHR.ts

```diff
@@ -0,0 +1,235 @@
+import xml2js from 'xml2js';
+import { Window } from 'happy-dom';
+
+export interface JobPost {
+    title: string;
+    description: string;
+    link: string;
+    jobURL: string;
+    vacancyDescription: string;
+    department: string;
+    salaryRange: string;
+    city: string;
+    slug: string;
+}
+
+export interface JobSummary {
+    title: string;
+    description: string;
+    department: string;
+    city: string;
+    slug: string;
+}
+
+async function fetchPeopleHRFeed(): Promise<string> {
```

Comment:
**Question**
I believe the current setup on `wagtail-torchbox` throws errors occasionally due to timeouts, it also heavily caches this endpoint. Due to the lack of peoplehr docs and the current `wagtail-torchbox` setup it seems like this endpoint might not be that reliable, so I'm not sure if SSR caching would be a great fit here. 

I wonder if caching this method would be a better approach i.e. somthing like this: https://stackoverflow.com/a/62005551

Thoughts @thibaudcolas ?

reference code from `wagtail-torchbox` below:

```py
# context_processors
def peoplehr_jobs_count(request):
    """
    Add the number of open jobs to the context
    """
    CACHE_KEY = "job_count"
    CACHE_TIMEOUT = 60 * 60 * 6

    job_count = cache.get(CACHE_KEY)

    if not job_count:
        job_index = JobIndexPage.objects.first()
        if job_index:
            peoplehr_feed = PeopleHRFeed()
            job_count = peoplehr_feed.get_job_count(job_index.jobs_xml_feed)
        else:
            job_count = 0
        cache.set(CACHE_KEY, job_count, CACHE_TIMEOUT)

    return {"job_count": job_count}
```
```py
# api.py
    def get_job_count(self, url=None):
        """
        Get the number of jobs listed on the current XML feed
        """
        if not url:
            return None

        try:
            resp = requests.get(url, timeout=3)

            resp.raise_for_status()
            xml_root = ElementTree.fromstring(resp.content)
            return len(xml_root.findall("channel/item"))
        except requests.exceptions.RequestException:
            logger.exception(f"Could not get People HR jobs feed from {url}")
            return None
        except requests.exceptions.Timeout:
            logger.exception(f"Timed out getting People HR jobs feed from {url}")
            return None
        except ElementTree.ParseError:
            logger.exception(f"Could not parse People HR jobs feed from {url}")
            return None
```

Reply:
I have no experience with `revalidate` myself but this feels pretty similar based on the docs. If we’re sure that the previous version of the page stays up when revalidation fails, I think that’s desirable for us in the case (there are little consequences to displaying outdated data).

Reply:
I've just had a moment of clarity in terms of what I was worried about before.

We'd need to check how NextJS handles query params in terms of caching the job detail page. it should just be the case of running this locally and making sure adding a query param will not overide caching i.e. cause an addtional peopleHR api request in [this line](https://github.com/torchbox/careers/pull/13/files/83d92b02c1580b9521a225f702dc14205731b8c6#diff-cbba9f194d1e0b9474ede31d264d1b5e526791293f08cf0fe1b88bb380a77026R21). 

Worth checking as I'd imagine a job posting board would be likely target for bots and I wouldn't want us querying to people HR too much as I'm not aware of our usage limits.

Reply:
ISR will handle generating the new posts for us, I've added caching so no-one will inadvertently DDOS the PeopleHR RSS feed with invalid URLs.